File: scripts/run_pipeline.py
#!/usr/bin/env python3
"""Entry point for the March Madness pipeline (Python)."""
import sys
from pathlib import Path

# Ensure project root is on path
project_root = Path(__file__).resolve().parent.parent
if str(project_root) not in sys.path:
    sys.path.insert(0, str(project_root))

# Change to project root
import os

os.chdir(project_root)

from src.pipelines.madness import run_pipeline

if __name__ == "__main__":
    run_pipeline()

File: scripts/run_pipeline.R
#!/usr/bin/env Rscript
# Entry point for the March Madness pipeline.
# Run from project root: Rscript scripts/run_pipeline.R

# Find project root and set working directory
if (file.exists("configs/default.yaml")) {
  proj_root <- getwd()
} else if (file.exists("../configs/default.yaml")) {
  proj_root <- normalizePath("..")
} else {
  stop("Run from project root or scripts/ directory. Expected configs/default.yaml")
}
setwd(proj_root)
message("Working directory: ", proj_root)

# Run the pipeline
source("src/pipelines/madness.R")
message("Pipeline complete.")
